### ISSUE TITLE
python error: object has no attribute 'encryption_password'

### ISSUE THREAD

**Issue by ato2** [archlinux/archinstall#1712]
Traceback (most recent call last):
  File "/usr/bin/archinstall", line 8, in <module>
    sys.exit(run_as_a_module())
  File "/usr/lib/python3.10/site-packages/archinstall/__init__.py", line 312, in run_as_a_module
    script.execute()
  File "/usr/lib/python3.10/site-packages/archinstall/lib/profiles.py", line 195, in execute
    self.spec.loader.exec_module(sys.modules[self.namespace])
  File "<frozen importlib._bootstrap_external>", line 883, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/usr/lib/python3.10/site-packages/archinstall/examples/guided.py", line 293, in <module>
    config_output = ConfigurationOutput(archinstall.arguments)
  File "/usr/lib/python3.10/site-packages/archinstall/lib/configuration.py", line 49, in __init__
    self._process_config()
  File "/usr/lib/python3.10/site-packages/archinstall/lib/configuration.py", line 75, in _process_config
    self._user_credentials['encryption_password'] = self._config[key].encryption_password
AttributeError: 'NoneType' object has no attribute 'encryption_password'
sudo archinstall  10.07s user 7.67s system 3% cpu 8:30.06 total

**Comment by ato2** [comment]
I didn't def. root pass. Installer should check it.

**Comment by Torxed** [comment]
I think the two are unrelated? But glad you got it working!

**Comment by ato2** [comment]
Still the same problem.  Current system, after pacman -Syu.

**Comment by Torxed** [comment]
Tbh, the easiest way to fix https://github.com/archlinux/archinstall/issues/1704 is to:
`nano +74 /usr/lib/python3.10/site-packages/archinstall/lib/configuration.py`
```diff
- if key == 'disk_encryption':
+ if key == 'disk_encryption' and self._config[key]:
```

**Comment by XaurDesu** [comment]
> Tbh, the easiest way to fix #1704 is to: `nano +74 /usr/lib/python3.10/site-packages/archinstall/lib/configuration.py`
> 
> ```diff
> - if key == 'disk_encryption':
> + if key == 'disk_encryption' and self._config[key]:
> ```
> 
> As the patch hasn't been released as a pacman package yet, which is why `pacman -Syu archinstall` won't help currently.

Couldn't we add this as a patch? Is there any reason why we can't?

**Comment by Torxed** [comment]
> > Tbh, the easiest way to fix #1704 is to: `nano +74 /usr/lib/python3.10/site-packages/archinstall/lib/configuration.py`
> > ```diff
> > - if key == 'disk_encryption':
> > + if key == 'disk_encryption' and self._config[key]:
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > As the patch hasn't been released as a pacman package yet, which is why `pacman -Syu archinstall` won't help currently.
> 
> Couldn't we add this as a patch? Is there any reason why we can't?

It already exists: https://github.com/archlinux/archinstall/pull/1705
